ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -322,14 +331,15 @@ private LogicalSchema rebuildWithoutPseudoAndKeyColsInValue() {
    * @return the number of columns added. This also functions as a "column index", to pass into
    * builder.add()
    */
-  private int addNonPseudoAndKeyColsToValueSchema(ImmutableList.Builder<Column> builder) {
+  private int addNonPseudoAndKeyColsToValueSchema(
+      final ImmutableList.Builder<Column> builder, final int pseudoColumnVersion) {
     final Map<Namespace, List<Column>> byNamespace = byNamespace();
 
     final List<Column> value = byNamespace.get(VALUE);
 
     int addedColumns = 0;
     for (final Column c : value) {
-      if (SystemColumns.isSystemColumn(c.name())) {
+      if (SystemColumns.isSystemColumn(c.name(), pseudoColumnVersion)) {
```

Comment:
Nice catch!

Reply:
The unit tests did the heavy lifting, but thanks :)